### PR TITLE
DBT-766 DBT-884 added online lesson class

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

This feature will provide students with the option of joining online lesson

[Task](https://www.notion.so/tianlu/Frontend-Access-to-the-online-lesson-page-permissions-ebe8bf6041a4449c90d1e6756e9bf062?pvs=4)

## Test

-  As an Admin Create an Online class and make sure you provide a Zoom url
- Assign a student to that lesson 
- As a student go to your classes and you should be able to see the join lesson button
- join the lesson

### CODE REVIEW

File: src/modules/lessons/student-join-online-clase/students-online-lesson.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import DataTableStore from '@/modules/resources/store/data-table.store'
+
+export default {
```
Comment (abelbordonado) [inline src/modules/lessons/student-join-online-clase/students-online-lesson.js:3]: I believe we dont require a store here. We are not keeping any information on it

File: src/modules/lessons/student-join-online-clase/students-online-lesson.vue [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      const regexPattern = /\/j\/(\d+)/
+      // Use RegExp.exec() to find the match
+      const [, meetingId] = regexPattern.exec(res.url) || []
+
+      this.meetingId = meetingId
+
+      if (meetingId) {
+        // Extracted meeting ID
+
+        console.log('Meeting ID:', meetingId)
```
Comment (abelbordonado) [inline src/modules/lessons/student-join-online-clase/students-online-lesson.vue:65]: 🦞

File: src/modules/lessons/student-join-online-clase/students-online-lesson.vue [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+/* each cell border in mobile */
+.v-data-table > .v-data-table__wrapper .v-data-table__mobile-row {
+  border-bottom: 2px solid #ededed;
+}
+
+.theme--light.v-divider {
+  border-color: rgba(0, 0, 0, 0.6);
+}
```
Comment (abelbordonado) [inline src/modules/lessons/student-join-online-clase/students-online-lesson.vue:93]: Do we need this CSS?

File: src/modules/lessons/student-join-online-clase/students-online-lesson.vue [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    </div>
+
+    <v-spacer></v-spacer>
+    <v-card style="height: 75vh;">
+      <section class="px-2 d-flex align-center">
+        <iframe
+          ref="zoomIframe"
+          allow="microphone; camera"
+          style="border: 0; height: 60vh; width: 100%;"
+          :src="'https://app.zoom.us/wc/' + meetingId + '/join'"
```
Comment (abelbordonado) [inline src/modules/lessons/student-join-online-clase/students-online-lesson.vue:23]: Used a computed property here, is more clean :D

File: src/modules/lessons/student-join-online-clase/index.vue [hunk 5]
```diff
@@ -0,0 +1,44 @@
+<template>
+  <v-card-text>
+    <div>
+      <v-card flat>
+        <v-card-text>
```
Comment (abelbordonado) [inline src/modules/lessons/student-join-online-clase/index.vue:5]: Im removing this ones, better for responsiveness.

File: src/modules/lessons/student-join-online-clase/index.vue [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  data() {
+    return {
+    }
+  },
+  mounted() {
+    this.loadNotifications()
+  },
+
+  beforeCreate() {
+    this?.$hasPermissionMiddleware(PermissionEnum.SEE_LESSONS) &&
```
Comment (abelbordonado) [inline src/modules/lessons/student-join-online-clase/index.vue:35]: No need the `?` after `this` , lol

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

File: src/modules/lessons/student-join-online-clase/students-online-lesson.vue (hunk 3 above)
Comment (kayiganwa04) [reply src/modules/lessons/student-join-online-clase/students-online-lesson.vue:93]: we don't I was trying to hide the vimeo logo

**Pull request merged**